spec/Section 3 -- Type System.md

```diff
@@ -2016,6 +2015,7 @@ TypeSystemDirectiveLocation : one of
 - `ENUM_VALUE`
 - `INPUT_OBJECT`
 - `INPUT_FIELD_DEFINITION`
+- `DOCUMENT`
```

Comment:
Should we extract document level directives to a separate spec PR, it seems to be lumped in here but sounds like a meaningful addition in the grander scheme either way

Reply:
Yeah truthfully we could also do no null propagation mode as it's own PR as well.

Reply:
These should indeed be separate PRs 👍